theano/gof/destroyhandler.py

```diff
@@ -114,43 +114,11 @@ def _contains_cycle(fgraph, orderings):
 
     # Pass through all the nodes to build visitable, parent_count, and
     # node_to_children
-    for var in fgraph.variables:
-
-        # this is faster than calling get_parents
-        owner = var.owner
-        # variables don't appear in orderings, so we don't need to worry
-        # about that here
-        if owner:
-            # insert node in node_to_children[r]
-            # (if r is not already in node_to_children,
-            # intialize it to [])
-            node_to_children.setdefault(owner, []).append(var)
-            parent_counts[var] = 1
-        else:
-            visitable.append(var)
-            parent_counts[var] = 0
-
-    for a_n in fgraph.apply_nodes:
-        parents = list(a_n.inputs)
-        # This is faster than conditionally extending
-        # IG: I tried using a shared empty_list = [] constructed
-        # outside of the for loop to avoid constructing multiple
-        # lists, but this was not any faster.
-        parents.extend(orderings.get(a_n, []))
-
-        if parents:
-            for parent in parents:
-                # insert node in node_to_children[r]
-                # (if r is not already in node_to_children,
-                # intialize it to [])
-                node_to_children.setdefault(parent, []).append(a_n)
-            parent_counts[a_n] = len(parents)
-        else:
-            # an Apply with no inputs would be a weird case, but I'm
-            # not sure we forbid it
-            visitable.append(a_n)
-            parent_counts[a_n] = 0
-
+    try:
+        graph.io_toposort(list(fgraph.variables), list(fgraph.apply_nodes))
+    except ValueError as ve:
+        if "cycles" in ve.args[0]:
+            return True
```

Comment:
@nouiz : I am trying to address the the speeding up of detecting cycles in graph. I followed what was given in line:65. With what i have done, I am getting this error : http://pastebin.com/ervsJM5p . Could you please point out where I am going wrong ?